dapp/src/router/index.tsx

```diff
@@ -32,6 +34,24 @@ export const router = createBrowserRouter([
   {
     path: "/",
     element: <Layout />,
+    loader: ({ request }) => {
+      // TODO: display the error page/modal when the referral is invalid.
```

Comment:
There is a case where we have a correct `ref` value in the URL, but then change it to an incorrect one. We still store the previous value in `localStorge`.  Shouldn't we also reset this value? 

<img width="653" alt="Screenshot 2024-11-22 at 09 01 19" src="https://github.com/user-attachments/assets/ba08739b-6828-4282-94b3-578aa2bf156f">

Reply:
aafe02f